src/blockchain/database/common/BlockFilter.cpp

```diff
@@ -359,54 +344,46 @@ auto BlockFilter::StoreFilters(
             BulkIndex>;
 
         // NOTE do as much work as possible before locking mutexes
-        static const auto options = [] {
-            auto out = google::protobuf::ArenaOptions{};
-            out.start_block_size = 8_MiB;
-            out.max_block_size = 8_MiB;
+        auto options = google::protobuf::ArenaOptions{};
+        options.start_block_size = 8_MiB;
+        options.max_block_size = 8_MiB;
 
-            return out;
-        }();
         auto arena = google::protobuf::Arena{options};
         auto upstream = alloc::StandardToBoost(alloc::System());
         auto alloc = alloc::BoostMonotonic{4_MiB, &upstream};
-        auto data = [&] {
-            auto out = Vector<StorageItem>{&alloc};
-            out.reserve(headers.size());
-            auto h = headers.cbegin();
-            auto f = filters.cbegin();
-
-            for (auto end = filters.cend(); end != f; ++f, ++h) {
-                auto& [bHash, cfHeader, cfilter, bytes, index] =
-                    out.emplace_back(
-                        BlockHash{},
-                        SerializedCfheader{&alloc},
-                        google::protobuf::Arena::Create<proto::GCS>(&arena),
-                        0,
-                        BulkIndex{});
-                bHash = std::get<0>(*h).Bytes();
-                const auto* cfheaderProto = [&] {
-                    const auto& [block, header, hash] = *h;
-                    auto* proto = google::protobuf::Arena::Create<
-                        proto::BlockchainFilterHeader>(&arena);
-                    proto->set_version(1);
-                    proto->set_header(header.data(), header.size());
-                    proto->set_hash(hash.data(), hash.size());
-
-                    return proto;
-                }();
-                proto::write(*cfheaderProto, writer(cfHeader));
-
-                if (auto& [b, filter] = *f;
-                    (false == filter.IsValid()) ||
-                    (false == filter.Internal().Serialize(*cfilter))) {
-                    throw std::runtime_error{"Failed to serialize gcs"};
-                }
 
-                bytes = cfilter->ByteSizeLong();
+        Vector<StorageItem> data{&alloc};
```

Comment:
I'd be inclined to take this fragment out to a private function and call it from here - the main logic is getting swamped by the gritty detail. Howerev, it's your choice, I won't insist.

Reply:
yeah, I generally think we have too big functnions there